src/eth-utils.js

```diff
@@ -1,5 +1,66 @@
 import { firmwareAPI, HARDENED } from './bitbox02';
 
+const getCoinFromChainId = chainId => {
+    switch(chainId) {
+        case 1:
+            return firmwareAPI.messages.ETHCoin.ETH;
+        case 3:
+            return firmwareAPI.messages.ETHCoin.RopstenETH;
+        case 4:
+            return firmwareAPI.messages.ETHCoin.RinkebyETH;
+        default:
+            throw new Error('Unsupported network');
+    }
+}
+
+/**
+ * @param keypathString keypath in string format e.g. m/44'/1'/0'/0
+ * @returns keypath as array e.g. [2147483692, 2147483649, 2147483648, 0]
+ */
+export const getKeypathFromString = keypathString => {
+    const levels = keypathString.toLowerCase().split('/');
+    if (levels[0] !== 'm') throw new Error('Invalid keypath');
+    return levels.flatMap(level => {
+        let hardened = false;
+        if (level === 'm' || level === '') {
+            return [];
+        }
+        if (level.substring(level.length - 1) === "'") {
+            hardened = true
+        }
+        level = parseInt(level);
+        if (isNaN(level) ||
+            level < 0 ||
+            (!hardened && level >= HARDENED) ||
+            (hardened && level >= HARDENED * 2)
```

Comment:
this check is still wrong i think; in `m/x'`, x still needs to be `< HARDENED`, as a couple of lines below you add `HARDENED` to it. so in both hardened and unhardened the check is the same here.